templates/taxi/manufacturer_list.html

```diff
@@ -23,11 +25,21 @@
               <td>
                   {{ manufacturer.country }}
               </td>
+              <td>
+                  <a class="btn btn-primary" href="{% url 'taxi:manufacturer-update' pk=manufacturer.id %}">Update</a>
+              </td>
+              <td>
+                  <a class="btn btn-primary" href="{% url 'taxi:manufacturer-delete' pk=manufacturer.id %}">Delete</a>
+              </td>
           </tr>
         {% endfor %}
       </table>
-
+  
     {% else %}
         <p>There are no manufacturers in the service.</p>
     {% endif %}
+  <br>
+  <a class="btn btn-primary" href="{% url 'taxi:manufacturer-create' %}">add new Manufacturer</a>
+  <a class="btn btn-primary" href="{% url 'taxi:index' %}">Cancel</a>
```

Comment:
Ensure that the 'Cancel' button leads to the previous page the user was on, as per the checklist requirement.